### PR TITLE
adds promotion model

### PR DESCRIPTION
**Pull request by bkudler**
Made the following changes:

Created Promotions model
Added methods for Promotion
Achieved 95% test coverage for model

### CODE REVIEW

**Review by tanaytrivedi — CHANGES_REQUESTED** [review]
I think you need to commit your new service.models file, because the update to that script is not in this commit.

File: tests/test_models.py [hunk 1]
```diff
@@ -1,37 +1,329 @@
...
+        promo.create()
+        logging.debug(promo)
+        self.assertIsNotNone(promo.id)
+        # Change it an save it
+        promo.end_date = date.today() + timedelta(2)
+        original_id = promo.id
+        promo.update()
+        self.assertEqual(promo.id, original_id)
+        self.assertEqual(promo.end_date, date.today() + timedelta(2))
+        # Fetch it back and make sure the id hasn't changed
```
Comment (loixe) [inline tests/test_models.py:119]: If we use "has_been_extended", we need update "has_been_extended" here as well?

File: tests/test_models.py [hunk 2]
```diff
@@ -1,37 +1,329 @@
...
+            promo.create()
+
+        promo = Promotion.find_or_404(promos[1].id)
+        self.assertIsNot(promo, None)
+        self.assertEqual(promo.id, promos[1].id)
+        self.assertEqual(promo.name, promos[1].name)
+        self.assertEqual(promo.start_date, promos[1].start_date)
+        self.assertEqual(promo.end_date, promos[1].end_date)
+        self.assertEqual(promo.whole_store, promos[1].whole_store)
+        self.assertEqual(promo.end_date, promos[1].end_date)
```
Comment (tanaytrivedi) [inline tests/test_models.py:323]: Duplicated end_date here too.

**Review by tanaytrivedi — CHANGES_REQUESTED** [review]
Duplicated end date here too.

File: tests/test_models.py (hunk 1 above)
Comment (bkudler) [reply tests/test_models.py:119]: good catch!

File: tests/test_models.py [hunk 3]
```diff
@@ -1,37 +1,329 @@
...
+        self.assertEqual(len(Promotion.all()), 5)
+        # find the 2nd promotion in the list
+        promo = Promotion.find(promos[1].id)
+        self.assertIsNot(promo, None)
+        self.assertEqual(promo.id, promos[1].id)
+        self.assertEqual(promo.name, promos[1].name)
+        self.assertEqual(promo.start_date, promos[1].start_date)
+        self.assertEqual(promo.end_date, promos[1].end_date)
+        self.assertEqual(promo.whole_store, promos[1].whole_store)
+        self.assertEqual(promo.end_date, promos[1].end_date)
```
[the remark this answers is not in the document]
Comment (bkudler) [reply tests/test_models.py:232]: fixed!

File: tests/test_models.py (hunk 2 above)
Comment (bkudler) [reply tests/test_models.py:323]: fixed

**Review by tanaytrivedi — APPROVED** [review]
All good now. Thanks.

**Review by loixe — APPROVED** [review]

**Pull request merged**